App crash on using share_plus package for sharing content as soon as you press share and dismiss the share sheet app crashed.

E/AndroidRuntime(16727): 	at android.app.ActivityThread.main(ActivityThread.java:8177)
E/AndroidRuntime(16727): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(16727): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
E/AndroidRuntime(16727): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
E/AndroidRuntime(16727): Caused by: kotlin.UninitializedPropertyAccessException: lateinit property resultChannel has not been initialized
E/AndroidRuntime(16727): 	at com.shirsh.flutter_doc_scanner.FlutterDocScannerPlugin.onActivityResult(FlutterDocScannerPlugin.kt:159)
E/AndroidRuntime(16727): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry$FlutterEngineActivityPluginBinding.onActivityResult(FlutterEngineConnectionRegistry.java:774)
E/AndroidRuntime(16727): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.onActivityResult(FlutterEngineConnectionRegistry.java:422)
E/AndroidRuntime(16727): 	at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onActivityResult(FlutterActivityAndFragmentDelegate.java:857)
E/AndroidRuntime(16727): 	at io.flutter.embedding.android.FlutterActivity.onActivityResult(FlutterActivity.java:884)
E/AndroidRuntime(16727): 	at android.app.Activity.dispatchActivityResult(Activity.java:8943)
E/AndroidRuntime(16727): 	at android.app.ActivityThread.deliverResults(ActivityThread.java:5520)
E/AndroidRuntime(16727): 	... 13 more
I/Process (16727): Sending signal. PID: 16727 SIG: 9
Lost connection to device.

--- a/android/src/main/kotlin/com/shirsh/flutter_doc_scanner/FlutterDocScannerPlugin.kt
+++ b/android/src/main/kotlin/com/shirsh/flutter_doc_scanner/FlutterDocScannerPlugin.kt
@@ -155,7 +155,7 @@ class FlutterDocScannerPlugin : MethodCallHandler, ActivityResultListener,
                 )
                 return@let true
             }
-        } else {
+        } else if (requestCode == REQUEST_CODE_SCAN || requestCode == REQUEST_CODE_SCAN_URI) {
             resultChannel.error("SCAN_FAILED", "Failed to start scanning", null)
         }
         return false

